### PR TITLE
implemented events for envvars and services

### PR DESCRIPTION
**Pull request by mihaibuzgau**

### CODE REVIEW

File: src/CloudFoundry.VisualStudio/Controls/EnvVarsUserControl.xaml.cs [hunk 1]
```diff
@@ -29,43 +29,44 @@ public EnvVarsUserControl()
 
         private void btnAddEnvVar_Click(object sender, RoutedEventArgs e)
         {
-            //var profile = (this.DataContext as PublishProfile);
+            var profile = (this.DataContext as PublishProfileEditorResources);
```
Comment (carabasdaniel) [inline src/CloudFoundry.VisualStudio/Controls/EnvVarsUserControl.xaml.cs]: You should check for null here before attempting to use the profile object

File: src/CloudFoundry.VisualStudio/Controls/EnvVarsUserControl.xaml.cs [hunk 2]
```diff
@@ -29,43 +29,44 @@ public EnvVarsUserControl()
...
-            //}
+            tbEnvVarValue.Clear();
+            tbEnvVarKey.Clear();
+            lvEnvVars.Items.Refresh();
         }
 
 
         private void btnRemoveEnvVar_Click(object sender, RoutedEventArgs e)
         {
+            var dataContext = (this.DataContext as PublishProfileEditorResources);
```
Comment (carabasdaniel) [inline src/CloudFoundry.VisualStudio/Controls/EnvVarsUserControl.xaml.cs]: Check for null before using dataContext

File: src/CloudFoundry.VisualStudio/Controls/EnvVarsUserControl.xaml.cs [hunk 3]
```diff
@@ -29,43 +29,44 @@ public EnvVarsUserControl()
...
-            //{
-            //    profile.CFAppManifest.EnvironmentVars.Add(tbEnvVarKey.Text, tbEnvVarValue.Text);
+            if (!profile.PublishProfile.Application.EnvironmentVariables.ContainsKey(tbEnvVarKey.Text))
+            {
+                profile.PublishProfile.Application.EnvironmentVariables.Add(tbEnvVarKey.Text, tbEnvVarValue.Text);
+
+            }
+            else
+            {
+                profile.PublishProfile.Application.EnvironmentVariables[tbEnvVarKey.Text] = tbEnvVarValue.Text;
```
Comment (viovanov) [inline src/CloudFoundry.VisualStudio/Controls/EnvVarsUserControl.xaml.cs]: You can just leave this line, no need to verify then add.

File: src/CloudFoundry.VisualStudio/Forms/PushDialog.xaml [hunk 4]
```diff
@@ -21,35 +21,34 @@
...
 
         <xctk:WizardPage x:Name="RoutesPage" PageType= "Interior"
-                         Title="Routes" Description="Routes" CanFinish="True">
+                         Title="Routes" Description="Application routes settings." CanFinish="True">
             <Controls:RoutesUserControl x:Name="ucRoutes" />
         </xctk:WizardPage>
 
         <xctk:WizardPage x:Name="ServicesPage" PageType= "Interior"
-                         Title="Services" Description="Services" CanFinish="True">
+                         Title="Services" Description="Applicaiton services settings." CanFinish="True">
```
Comment (viovanov) [inline src/CloudFoundry.VisualStudio/Forms/PushDialog.xaml]: Typo Applicaiton

**Comment by viovanov** [comment]
:+1:

**Pull request merged**